### ISSUE TITLE
NuGet restore fails due to alleged 'cycle detected' (.NET SDK 7.0 regression)

### ISSUE THREAD

**Issue by AArnott** [NuGet/Home#12342]
### NuGet Product Used

dotnet.exe

### Product Version

7.0.101

### Worked before?

6.0.100

### Impact

I'm unable to use this version

## Repro steps

```
git clone https://github.com/AArnott/CSharpIsNull.git
cd CSharpIsNull
git checkout libtemplate
dotnet restore
```

Expected success, but got this:

```
C:\Users\andarno\source\repos\CSharpIsNull\src\CSharpIsNullAnalyzer.CodeFixes\CSharpIsNullAnalyzer.CodeFixes.csproj : error NU1108: Cycle detected.  [C:\Users\andarno\source\repos\CSharpIsNull\CSharpIsNullAnalyzer.sln]
C:\Users\andarno\source\repos\CSharpIsNull\src\CSharpIsNullAnalyzer.CodeFixes\CSharpIsNullAnalyzer.CodeFixes.csproj : error NU1108:   CSharpIsNullAnalyzer -> CSharpIsNullAnalyzer (>= 0.1.329). [C:\Users\andarno\source\repos\CSharpIsNull\CSharpIsNullAnalyzer.sln]
```

Diagnostic logs: [msbuild.log](https://github.com/NuGet/Home/files/10339480/msbuild.log)

### Verbose Logs

```shell
See logs attachment in repro steps
```

**Comment by kartheekp-ms** [comment]
I am able to reproduce the issue following the steps mentioned in the issue description.

**Comment by aortiz-msft** [comment]
Triage: @jeffkl - would you please help us understand the severity of this issue?

**Comment by jeffkl** [comment]
@AArnott this is being caused by your `<GlobalPackageReference />` to the `CSharpIsNullAnalyzer` package [here](https://github.com/AArnott/CSharpIsNull/blob/libtemplate/Directory.Packages.props#L21).  The `CSharpIsNullAnalyzer` project cannot reference a project or package with the same name.  The easiest way around it is to disable this `GlobalPackageReference` for the one project.

I pushed a commit in a fork that shows my recommendation on adding a property to disable a GlobalPackageReference for any particular project in the tree:

https://github.com/jeffkl/CSharpIsNull/commit/6aea7ab81c31889edef5390ddb5e479c36f56bca

Of course this project will not use the package which might not be desired.  This is a duplicate of https://github.com/NuGet/Home/issues/6754 which has a workaround that I've also used: https://github.com/NuGet/Home/issues/6754#issuecomment-388166770
